frontend/src/ui-kit/components/ErrorBoundary/ErrorBoundary.test.tsx

```
@@ -0,0 +1,289 @@
+        </ErrorBoundary>
+      );
+
+      expect(screen.getByTestId('custom-fallback')).toBeInTheDocument();
+      expect(screen.getByText('Custom error UI')).toBeInTheDocument();
+    });
+
+    it('should support fallback render function', () => {
+      const fallbackRender = vi.fn(({ error, resetErrorBoundary }) => (
+        <div data-testid="fallback-render">
+          <span data-testid="error-msg">{error.message}</span>
+          <button onClick={resetErrorBoundary}>Reset</button>
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)